docker/Dockerfile

```diff
@@ -0,0 +1,130 @@
+FROM ubuntu:16.04
+
+MAINTAINER Daniel Baptista Dias <danielbpdias@gmail.com>
+
+ENV GENSIM_REPOSITORY https://github.com/RaRe-Technologies/gensim/archive
+ENV GENSIM_VERSION master
+
+# Installs python, pip and setup tools (with fixed versions)
+RUN apt-get update \
+    && apt-get install -y \
+    ant=1.9.6-1ubuntu1 \
+    cmake=3.5.1-1ubuntu3 \
+    default-jdk=2:1.8-56ubuntu2 \
+    g++=4:5.3.1-1ubuntu1 \
+    git=1:2.7.4-0ubuntu1 \
+    libboost-all-dev=1.58.0.1ubuntu1 \
+    libgsl-dev=2.1+dfsg-2 \
+    mercurial=3.7.3-1ubuntu1 \
+    python3=3.5.1-3 \
+    python3-pip=8.1.1-2ubuntu0.4 \
+    python3-setuptools=20.7.0-1 \
+    unzip=6.0-20ubuntu1 \
+    wget=1.17.1-1ubuntu1.1 \
+    libboost-program-options-dev \
+    libboost-python-dev \
+    zlib1g-dev
+
+# Setup python language
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+ENV LC_CTYPE en_US.UTF-8
+ENV LC_ALL en_US.UTF-8
+
+# Make some symlinks
+RUN cd /usr/bin \
+    && ln -s easy_install3 easy_install \
+    && ln -s pip3 pip
+
+# Upgrade pip
+RUN pip install --upgrade pip
+
+# Install dependencies
+RUN pip install \
+    annoy \
+    cython \
+    jupyter \
+    matplotlib \
+    nltk \
+    pandas \
+    pyemd \
+    sklearn \
+    testfixtures \
+    unittest2 \
+    git+git://github.com/mila-udem/fuel.git@stable \
+    git+git://github.com/mila-udem/blocks.git@stable \
+    -r https://raw.githubusercontent.com/mila-udem/blocks/stable/requirements.txt
+
+# Create gensim directory and dependencies directory
+RUN mkdir /gensim \
+    && mkdir /gensim_dependencies
+
+# Download gensim from Github
+RUN mkdir /gensim/download \
+    && cd /gensim/download \
+    && wget --quiet $GENSIM_REPOSITORY/$GENSIM_VERSION.zip \
+    && unzip $GENSIM_VERSION.zip \
+    && mv ./gensim-$GENSIM_VERSION/* /gensim \
+    && rm -rf /gensim/download \
+    && cd /gensim \
+    && python setup.py install
```

Comment:
It will already download the package from branch which is specified in GENSIM_VERSION variable (and I use a branch on my personal fork for now which has the docker folder).